### PR TITLE
Remove native code info from xpack info api

### PR DESCRIPTION
**Pull request by rjernst**
The native code info is a portion of the xpack info api that emits
arbitrary info about feature plugins that are backed by native code.
This is currently only used by machine learning. Additionally, it is the
only non enabled/available information reported by the info api.

This commit moves the native code info reporting to the usage api for
machine learning. The commit info reported is only used for debugging
purposes, and not captured by the current uses of the info api
(monitoring and telemetry) since it requires passing the verbose flag.

Long term, this information would be better suited to node info. Until
node info is extendable by plugins, this new location will allow
cleaning up the xpack info api to be implemented in a generic way for
all xpack features.

### CODE REVIEW

**Comment by droberts195** [comment]
> Additionally, it is the only non enabled/available information reported by the info api.

The info API also reports the build date and Git commit hash for X-Pack, for example the output might start with:

```
{
  "build" : {
    "hash" : "d3b3bcd",
    "date" : "2019-06-12T08:53:20.478372Z"
  },
```

Since the native code info is equivalent information this is why we originally added it to this endpoint.  (Having the build hash and date in X-Pack info is unnecessary since 6.3 as it's the same as for the main Elasticsearch server, but that's the history of why the ML native info was there in the first place.)

> The commit info reported is only used for debugging purposes, and not captured by the current uses of the info api (monitoring and telemetry) since it requires passing the verbose flag.

In reality everybody calling the info endpoint sees the verbose info, as it [piggy backs on `human` with a default of `true`](https://github.com/elastic/elasticsearch/blob/f5fde1d0843d2f0f53d3b9a15b9cfc8b94471ab7/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/RestXPackInfoAction.java#L39), so you'd have to explicitly add `human=false` to your request not to see it and I cannot imagine many people do that.

The other (more important) point is that monitoring and telemetry call the usage endpoint, not the info endpoint.  So this change is moving the native code info _into_ the monitoring and telemetry data.

> Long term, this information would be better suited to node info

This means that ML native code info will be in the monitoring and telemetry data for a few versions between now and when that change is made, and will then disappear.

Therefore I wonder if a better temporary home might be the [`MlInfoAction`](https://github.com/elastic/elasticsearch/blob/f5fde1d0843d2f0f53d3b9a15b9cfc8b94471ab7/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/MlInfoAction.java)?

Or if you leave the X-Pack usage endpoint as the temporary home then we need to get confirmation from someone who knows more than I do about monitoring and telemetry that adding some fields for a few versions and then removing them will not cause any problems.

**Comment by rjernst** [comment]
I did not realize the default for verbose was true. I'm happy to implement your suggestion of using the ml info api (I didn't know it existed). I've opened https://github.com/elastic/elasticsearch/pull/43172 to begin that. The plan would be to first add it there, with backporting to 7.x, and then remove native code info from master. I will update this PR to be just a pure removal of native code info from feature sets.

**Review by jaymode — APPROVED** [review]
LGTM

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/docs-check

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleFeatureSet.java [hunk 1]
```diff
@@ -61,7 +61,6 @@ public boolean enabled() {
         return enabled;
     }
 
-    @Override
     public Map<String, Object> nativeCodeInfo() {
```
Comment (droberts195) [inline x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleFeatureSet.java]: Can you remove this whole method?

**Review by droberts195 — APPROVED** [review]
LGTM, but I think there is one more place where you can remove a `nativeCodeInfo()` method.

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Pull request merged**